pkg/webreg/webreg.go

```diff
@@ -1880,37 +1879,310 @@ as the component.
+and there are checks in place enforcing that the repo stays consistent w.r.t.
+all this automation. When these checks fail, they usually advise how to run the
+tooling (using containers) to bring the repo to the desired state:
+</p>
+
+{{ plaintextSyntax (index . "determinizeCheckExample") }}
+
+<p>
+While there are individual <code>make</code> targets for different parts of the
+repository, it is easiest to run the <code>make update</code> that runs <em>all</em>
+these tools before a pull request submission:
+</p>
+
+{{ plaintextSyntax (index . "makeUpdateExample") }}
+
+<h5 id="rehearsals"><a href="#rehearsals">Rehearsals</a></h5>
+
+<p>
+In addition to the "normal" checks executed against pull requests on <code>openshift/release</code>,
+so-called <em>"rehearsals"</em> are triggered whenever a pull request would
+affect one or more CI jobs. Jobs that would be affected by such PR are executed
+as if run against a target component repository after the changes would be
+merged. This provides pull request authors early feedback about how config
+changes impact CI setup.
+</p>
+
+<p>
+All pull requests trigger a <code>ci/prow/pj-rehearse</code> job which inspects
+the changes in the PR and detects affected jobs. It then submits these jobs for
+execution and they will report to the pull request results via the GitHub
+contexts named with the <code>ci/rehearse/$org/$repo/$branch/$test</code>
+pattern. Neither the "driver" job (<code>ci/prow/pj-rehearse</code>) nor the
+individual rehearsals do not block merges.
+</p>
+
+<p>
+The following changes are considered when triggering rehearsals:
+</p>
+
+<ol>
+    <li>Changes to a Prow jobs themselves (<code>ci-operator/jobs</code>)</li>
+    <li>Changes to<code>ci-operator</code> configuration files(<code>ci-operator/config</code>)</li>
+    <li>Changes to multi-stage steps (<code>ci-operator/step-registry</code>)</li>
+    <li>Changes to templates (<code>ci-operator/templates</code>)</li>
+    <li>Changes to cluster profiles (<code>cluster/test-deploy</code>)</li>
+</ol>
+
+<p>
+The affected jobs are further filtered down so that jobs are only rehearsed when
+it is safe to do so. Only the jobs with <code>pj-rehearse.openshift.io/can-be-rehearsed: "true"</code>
+label are rehearsed. Because most of the jobs are generated, the presence of
+this label is usually set by the generator. By default, all generated presubmits
+and periodics are rehearsable, and postsubmit are not (because postsubmit are
+often used for publishing and similar actions not suitable to be executed in a
+rehearsal). Hand-crafted jobs can opt to be rehearsals by including this label.
+</p>
+
+<p>
+It is not possible to re-run individual rehearsal jobs. They do not react to any
+trigger commands. Rerunning rehearsals must be done by rerunning the "driver"
+job: <code>ci/prow/pj-rehearse</code>, which then triggers all rehearsals of
+jobs currently affected by the PR, including the rehearsals that passed before.
```

Comment:
"Why \<job that was selected based on sampling> is not retested?" is also a recurring question.

Reply:
True! I added a paragraph on that.